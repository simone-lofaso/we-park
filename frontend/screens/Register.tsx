import { Button, Input, Text } from '@rneui/themed';
import { useState } from 'react';
import { ImageBackground, SafeAreaView, StyleSheet } from 'react-native';
import { doRegister } from '../api';
import NorthGarage from '../assets/images/north-garage.jpg';
import { RootStackScreenProps } from '../types';

type FormState = {
  email: string;
  password: string;
};

export default function Register({
  navigation,
}: RootStackScreenProps<'Register'>) {
  const [form, setForm] = useState<FormState>({
    email: '',
    password: '',
  });

  return (
    <SafeAreaView style={styles.container}>
      <ImageBackground source={NorthGarage} style={styles.picture}>
        <Text style={styles.register}>Register</Text>
        <Input
          placeholder='Email'
          onChangeText={(e) => setForm({ ...form, email: e })}
          inputStyle={styles.input}
          placeholderTextColor='white'
        />
        <Input
          placeholder='Password'
          onChangeText={(e) => setForm({ ...form, password: e })}
          inputStyle={styles.input}
          placeholderTextColor='white'
        />
        <Button title='Submit' onPress={() => doRegister(form, navigation)} />
        <Button
          title='Go to Login'
          onPress={async () => {
            navigation.navigate('Login');
          }}
        />
      </ImageBackground>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    padding: 20,
    width: '100%',
  },
  picture: {
    alignItems: 'center',
    height: '100%',
    justifyContent: 'center',
    width: '100%',
  },
  register: {
    color: 'white',
    fontSize: 20,
  },
  input: {
    color: 'white',
  },
});
