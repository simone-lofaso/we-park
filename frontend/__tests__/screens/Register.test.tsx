import React from 'react';
import { create } from 'react-test-renderer';
import Register from '../../screens/Register';

it(`renders correctly`, () => {
  const props: any = {};

  const tree = create(<Register {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
