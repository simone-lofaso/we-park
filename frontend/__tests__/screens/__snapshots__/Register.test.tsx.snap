// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Text
      accessibilityRole="text"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    >
      Register
    </Text>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          editable={true}
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="white"
          style={
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          editable={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="white"
          style={
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Go to Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
